File: src/main/java/com/adjorno/billib/rest/db/TrendType.java
package com.adjorno.billib.rest.db;

import javax.persistence.*;

@Entity
@Table(name = "TREND_TYPE")
public class TrendType {
    public static final long TYPE_ALL = 0;
    public static final long TYPE_GAINERS = 1;
    public static final long TYPE_DEBUTS = 2;
    public static final long TYPE_FUTURES = 3;
    public static final long TYPE_SENIORS = 4;

    @Id
    @Column(name = "_ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long mId;

    @Column(name = "DESCRIPTION")
    private String mDescription;

    public TrendType() {
    }

    public TrendType(Long id, String description) {
        mId = id;
        mDescription = description;
    }

    public Long getId() {
        return mId;
    }

    public void setId(Long id) {
        mId = id;
    }

    public String getDescription() {
        return mDescription;
    }

    public void setDescription(String description) {
        mDescription = description;
    }
}

File: src/main/java/com/adjorno/billib/rest/db/TrendType.kt
package com.adjorno.billib.rest.db

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "TREND_TYPE")
data class TrendType(
        @Id
        @Column(name = "_id")
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @Column(name = "DESCRIPTION")
        var description: String? = null
) {
    companion object {
        const val TYPE_ALL: Long = 0
        const val TYPE_GAINERS: Long = 1
        const val TYPE_DEBUTS: Long = 2
        const val TYPE_FUTURES: Long = 3
        const val TYPE_SENIORS: Long = 4
    }
}
